libs/remix-ui/run-tab/src/lib/run-tab.tsx

```diff
@@ -235,6 +235,16 @@ export function RunTabUI(props: RunTabProps) {
     }
   }, [runTab.popup])
 
+  useEffect(() => {
+    if (runTab.selectExEnv.includes('injected') &&
+      Object.entries(runTab.accounts.loadedAccounts).length === 0 &&
+    runTab.accounts.selectedAccount.length > 0) {
+      // switch to vm-cancum because no account is loaded from injected provider
+      const context = plugin.blockchain.defaultPinnedProviders[0] // vm-cancun
```

Comment:
can we load load the previously selected or it will be difficult to do?

Reply:
It would just require some more code to track any account that is selected and persist it outside the reach of a useEffect.